neuralforecast/core.py

```diff
@@ -597,16 +597,23 @@ def get_missing_future(
         return ufp.anti_join(expected, futr_df[ids], on=ids)
 
     def _get_needed_futr_exog(self):
-        return set(
-            chain.from_iterable(getattr(m, "futr_exog_list", []) for m in self.models)
-        )
+        futr_exogs = []
+        for m in self.models:
+            futr_exogs += getattr(m, "futr_exog_list", [])
+            if len(getattr(m, "config", {})):
```

Comment:
This probably won't work for optuna, I'll think on how to do this

Reply:
ah right yes I was just testing with ray - I'm not very familiar with Optuna so happy to modify according to suggestions

Reply:
Hey. Sorry for taking so long. I think we can modify the MockTrial to return the choices
https://github.com/Nixtla/neuralforecast/blob/957a0ccb34d17395ad1da15ef1b027f8dc2c1944/neuralforecast/common/_base_auto.py#L23-L24
i.e. change it to:
```python
def suggest_categorical(self, name, choices):
    return choices
```
Then when trying to get the exogs we check if the config is a dict (ray) of if it's a callable (optuna)
* If it's ray we check if the features are being tuned, if they're just a list we extract them directly, otherwise the sample has a `categories` attribute that holds the possible values.
* If it's optuna we use the `MockTrial` to get the possible values and then extract them directly.

I believe something like this should work, but feel free to reach out if you get stuck:
```python
from neuralforecast.common._base_auto import BaseAuto, MockTrial

for m in self.models:
    if isinstance(m, BaseAuto):
        if isinstance(m.config, dict):  # ray
            exogs = m.config.get('futr_exog_list', [])
            if hasattr(exogs, 'categories'):  # features are being tuned, get possible values
                exogs = exogs.categories
        else:   # optuna
            exogs = m.config(MockTrial()).get('futr_exog_list', [])
    else:  # regular model, extract them directly
```

Reply:
I've updated those scripts and that seems to work ok - I've also updated the tests to optuna to confirm this !